File: next-js-boilerplate/src/validators/forms/billing.ts
import { z } from "zod";

export const billingSchema = z.object({
  plan: z.string().min(1),
  billingPeriod: z.enum(["monthly", "yearly"]),
  paymentMethod: z.string().min(1),
  couponCode: z.string().optional(),
  taxId: z.string().optional(),
});

export function createBillingFieldSchemas(t: Record<string, string>) {
  return {
    plan: z.string().min(1, t.planRequired ?? "Plan is required"),
    billingPeriod: z.enum(["monthly", "yearly"], {
      message: t.periodRequired ?? "Billing period is required",
    }),
    taxId: z.string().optional(),
  };
}

File: flutter-boilerplate/lib/validators/forms/billing.dart
String? validatePlan(String? value) {
  if (value == null || value.isEmpty) return 'Plan is required';
  return null;
}

String? validateBillingPeriod(String? value) {
  if (value == null || value.isEmpty) return 'Billing period is required';
  if (!['monthly', 'yearly'].contains(value)) return 'Invalid billing period';
  return null;
}

String? validatePaymentMethod(String? value) {
  if (value == null || value.isEmpty) return 'Payment method is required';
  return null;
}

String? validateCouponCode(String? value) {
  return null;
}
